### PR TITLE
"Save and Exit"/"Publish" actions in wizard

### LINKED ISSUE #66: Implement "save and exit" and "publish" actions in wizard
#### Proposed solution

Save and exit and publish currently just log trivial messages to console.

#### DoD (Definition of Done)

Actions are implemented

### PR DESCRIPTION
**Pull request by csc-jm**
### Description

PR adds more functionality to the footer of the wizard with the "Save and Exit" and "Publish" buttons. "Cancel" was also edited as I noticed an unnecessary alert pop-up with it in the Wizard frontpage step.

### Related issues

Fixes #66 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- modified save/publish alerts
- modified the conditionals for the actions of cancel/save/exist buttons
- added a patch operator to set folder `published` key to `true`

### Testing

- [x] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

**Comment by lilachic** [comment]
Should this be PR to develop branch instead of master? How did we wanted this to flow? From https://github.com/CSCfi/metadata-submitter-frontend/blob/master/CONTRIBUTING.md#submitting-changes it is develop?

**Comment by csc-jm** [comment]
> Should this be PR to develop branch instead of master? How did we wanted this to flow? From https://github.com/CSCfi/metadata-submitter-frontend/blob/master/CONTRIBUTING.md#submitting-changes it is develop?

Oh yea, definitely.

**Comment by csc-jm** [comment]
Also as pointed out by @saulipurhonen. Publishing here errors with: `"detail: "Request contains '/published' key that cannot be updated to folders."`

This is because of this line in the backend:
https://github.com/CSCfi/metadata-submitter/blob/b928909fbfc7635f711703e2999cb2539c81a33a/metadata_backend/api/handlers.py#L358

The line should be changed to: `allowed_paths = ["/name", "/description", "/metadataObjects", "/published"]` to work

**Comment by blankdots** [comment]
@csc-jm @saulipurhonen we can enable `published` there, but would be nice to keep track of this workflow.

We did remove it for a reason just don't remember why. I will enable it.

**Review by blankdots — COMMENTED** [review]
Does this PR just opens some dialog boxes upon those save actions as I see no API calls ?

File: src/features/wizardSubmissionFolderSlice.js [hunk 1]
```diff
@@ -65,6 +65,17 @@ export const addObjectToFolder = (folderID: string, objectDetails: ObjectInFolde
   dispatch(addObject(objectDetails))
 }
 
+export const publishFolderContent = (folder: Folder) => async (dispatch: any => void) => {
+  const changes = [{ op: "replace", path: "/published", value: true }]
+  const response = await folderAPIService.patchFolderById(folder.id, changes)
+  if (!response.ok) console.error(`Couldn't publish folder with id ${folder.id}`)
+  if (!response.ok) {
+    console.log(response)
```
Comment (blankdots) [inline src/features/wizardSubmissionFolderSlice.js:73]:
```suggestion
```
log is for debugging

File: src/components/NewDraftWizard/WizardComponents/WizardAlert.js [hunk 2]
```diff
@@ -93,7 +93,7 @@ const CancelFormDialog = ({
           dialogContent = "Folder has been saved"
           dialogActions = (
             <DialogActions style={{ justifyContent: "center" }}>
-              <Link component={RouterLink} aria-label="Cancel a new folder and move to frontpage" to="/">
+              <Link component={RouterLink} aria-label="Save a new folder and move to frontpage" to="/home">
```
[the remark this answers is not in the document]
Comment (saulipurhonen) [reply src/components/NewDraftWizard/WizardComponents/WizardAlert.js:96]: I think this should pass without rebase. Just commited merge fix.

**Comment by saulipurhonen** [comment]
Tested with "/published" path on backend and seems to work great.

**Review by saulipurhonen — APPROVED** [review]

**Review by blankdots — APPROVED** [review]

**Pull request merged**